presto-main/src/main/java/com/facebook/presto/Session.java

```diff
@@ -381,7 +343,7 @@ public SessionRepresentation toSessionRepresentation()
                 startTime,
                 systemProperties,
                 connectorProperties,
-                roles,
+                identity.getRoles(),
```

Comment:
Yeah. Session representation is just a way we serialize session. Identity is not serializable. So, we pass identity field by field.